src/core/lis-modal-element.ts

```diff
@@ -0,0 +1,146 @@
+ * {@link name | `name`}, {@link heading | `heading`}, and
+ * {@link content | `content`} attributes/properties can be set via HTML or
+ * javascript.
+ *
+ * For example:
+ * ```html
+ * <!-- set the name, heading and content via HTML -->
+ * <lis-modal-element
+ *   name="modal-test"
+ *   heading="Test Modal"
+ *   content="<p>Some HTML or text to be rendered</p>">
+ * </lis-modal-element>
+ * ```
+ *
+ * @example
+ * In the example below, the lis-simple-table-element web component
+ * is rendered within the lis-modal-element.
+ *
+ * The attributes/properties for lis-simple-table-element are set below
+ * in javascript. Please see the documentation for lis-simple-table-element
+ * for more information.
+ *
+ * ```html
+ * <lis-modal-element 
+ *   name="modal-test" 
+ *   heading="Cheesy Table Modal"
+ *   content="
+ *   <lis-simple-table-element 
+ *   id='table'>
+ *   </lis-simple-table-element>">
+ * </lis-modal-element>
+ * 
+ * <script type="text/javascript">
+ *  // get the simple table element after page loads.
+ *  window.onload = (event) => {
+ *   console.log("page is fully loaded");
+ *
+ *   const tableElement = document.getElementById('table');
+ *   // set the element's properties
+ *   tableElement.caption = 'My cheesy table';
+ *   tableElement.dataAttributes = ['cheese', 'region'];
+ *   tableElement.header = {cheese: 'Cheese', region: 'Region'};
+ *   tableElement.data = [
+ *     {cheese: 'Brie', region: 'France'},
+ *     {cheese: 'Burrata', region: 'Italy'},
+ *     {cheese: 'Feta', region: 'Greece'},
+ *     {cheese: 'Gouda', region: 'Netherlands'},
+ *   ];
+ *  }
+ * </script>
+ * ```
+ */
+@customElement('lis-modal-element')
+export class LisModalElement extends LitElement {
+
+  /** @ignore */
+  // used by Lit to style the Shadow DOM
+  // not necessary but exclusion breaks TypeDoc
+  static override styles = css``;
+
+  /** @ignore */
+  // disable Shadow DOM to inherit global styles
+  override createRenderRoot() {
+    return this;
+  }
+
+  /**
+   * The text to use as the Id for the uk-modal.
+   * This is used to bind buttons to show/hide.
+   *
+   * @attribute
+   */
+  @property({type: String})
+  name: string = "lis-modal";
+
+  /**
+   * The text or HTML to populate uk-modal-header
+   *
+   * @attribute
+   */
+  @property({type: String})
+  heading: string = "";
+
+  /**
+   * The text or HTML to populate uk-modal-body
+   *
+   * @attribute
+   */
+  @property({type: String})
+  content: string = "";
+
+  /** @ignore */
+  // returns the modal heading portion of the component
+  private _getHeading() {
+    if (!this.heading) {
+      return html``;
+    }
+    return html`<div class="uk-modal-header">${unsafeHTML(this.heading)}</div>`;
+  }
+
+  /** @ignore */
+  // returns the content provided in the modal-body
+  private _getContent() {
+    if (!this.content) {
+      return html``;
+    }
+    return html`<div class="uk-modal-body" uk-overflow-auto>${unsafeHTML(this.content)}</div>`;
```

Comment:
@ctcncgr I found [a workaround](https://github.com/lit/lit-element/issues/553#issuecomment-1217985861) that works if you only need one slot!
```typescript
return html`<div class="uk-modal-body" uk-overflow-auto>${Array.from(this.children)}</div>`;
```
Then you can use the modal element's as if it had a slot
```html
<lis-model-element>
    <lis-simple-table-element></lis-simple-table-element>
</lis-model-element>
```
Note that the solution I linked doesn't use `Array.from`; I found that you have to convert `this.children` (an [HTMLCollection](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection)) into an array otherwise Lit will only put the first child element in the modal and the other children will be placed outside of the modal, i.e. they won't be in the modal and you'll see them when the modal isn't visible.

Reply:
This is actually a cool way to emulate the `slot` tag, and named slots maybe, without the shadow dom. We could iterate through all the children and use their `name` attribute to build specific parts of a component. Unnamed children would be considered part of a general unnamed `slot`. Cool idea for future stuff.

Reply:
So I played around some more with this pseudo-slot and apparently if the Shadow DOM is disabled then the `slot="target-slot-name"` attributes are preserved in the Lit element's children collection. This means we could emulate named slots as you described but using the actual named slot syntax! This would be really cool because we would basically be providing a drop-in replacement for the slot API, i.e. the syntax is in line with the Web Standard.

Either way, this would make a super useful controller for our library.